src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs

```diff
@@ -173,10 +173,14 @@ private void OnTimer(object sender, ElapsedEventArgs e)
                 return;
             }
 
-            // restart the timer with the next schedule occurrence
             await InvokeJobFunction(DateTime.Now, false);
 
-            StartTimer(DateTime.Now);
+            // Restart the timer with the next schedule occurrence, but only 
+            // if Cancel, Stop, and Dispose have not been called.
+            if (!_cancellationTokenSource.IsCancellationRequested)
```

Comment:
Yeah that should work too.

---
In reply to: [78818151](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/149#discussion_r78818151) [](ancestors = 78818151)